app/services/analytics_service.rb

```diff
@@ -79,26 +69,26 @@ def stats_grouped_by_day
 
   private
 
-  attr_reader :options, :user_or_org, :article_data, :article_ids, :reaction_data, :comment_data, :follow_data, :page_view_data
+  attr_reader :user_or_org, :start_date, :end_date, :article_data, :reaction_data, :comment_data, :follow_data, :page_view_data
 
-  def start_date
-    options[:start].to_datetime.beginning_of_day
-  end
+  def load_data
+    @article_data = Article.where("#{user_or_org.class.name.downcase}_id" => user_or_org.id, published: true)
+    article_ids = @article_data.pluck(:id)
 
-  def end_date
-    if options[:end].present?
-      options[:end].to_datetime.end_of_day
+    if @start_date && @end_date
+      @reaction_data = Reaction.where(reactable_id: article_ids, reactable_type: "Article").
+        where(created_at: @start_date..@end_date).
```

Comment:
Also did not know you could use a range there!

Reply:
I had to google that, I knew it could be possible somehow, didn't remember the exact syntax, turns out you just have to give it a range and Rails does the magic